context/src/main/java/io/grpc/Deadline.java

```diff
@@ -231,6 +232,35 @@ public int compareTo(Deadline that) {
     return 0;
   }
 
+  @Override
+  public int hashCode() {
+    return Arrays.asList(this.ticker, this.deadlineNanos, this.isExpired()).hashCode();
+  }
+
+  @Override
+  public boolean equals(final Object o) {
+    if (o == this) {
+      return true;
+    }
+    if (!(o instanceof Deadline)) {
+      return false;
+    }
+
+    final Deadline other = (Deadline) o;
+    final Object this$ticker = this.ticker;
+    final Object other$ticker = other.ticker;
+    if (this$ticker == null ? other$ticker != null : !this$ticker.equals(other$ticker)) {
+      return false;
+    }
+    if (this.deadlineNanos != other.deadlineNanos) {
+      return false;
+    }
+    if (this.isExpired() != other.isExpired()) {
```

Comment:
Please do not include `isExpired()`, otherwise a dealine may not even equal to itself.

Reply:
Ah makes sense. At the time of expiry this might cause failure. WIll remove it